src/gui/parameters_screen.py

```diff
@@ -0,0 +1,413 @@
+                        return True
+                except ValueError:
+                    pass
+                self.point_side_size_validation_label.config(text="Invalid value", fg="red")
+                self.confirm_and_continue.config(state=DISABLED)
+                self.correctly_set_parameters[9] = 0
+            return False
+
+        def validate_all_parameters(self):
+            self.validate_coordinates_top()
+            self.validate_coordinates_down()
+            self.validate_coordinates_left()
+            self.validate_coordinates_right()
+            self.validate_time_range_start()
+            self.validate_time_range_end()
+            self.validate_data_time_step()
+            self.validate_cells_side_count_latitude()
+            self.validate_cells_side_count_longitude()
+            self.validate_point_side_size()
+
+        def check_all_parameters_validity_and_refresh_image(self):
+            if sum(self.correctly_set_parameters[:4]) == 4:
+                self.load_and_crop_image()
+                if sum(self.correctly_set_parameters) == 10:
+                    self.confirm_and_continue.config(state=NORMAL)
+
+        def browse_button(self):
+            filename = filedialog.askdirectory()
+            if filename:
+                self.data_path.set(filename)
+
+        def confirm_and_start_simulation(self):
+            set_simulation_coordinates_parameters(self.top_coord,
+                                                  self.down_coord,
+                                                  self.left_coord,
+                                                  self.right_coord,
+                                                  self.time_range_start,
+                                                  self.time_range_end,
+                                                  self.data_time_step_minutes,
+                                                  self.cells_side_count_latitude,
+                                                  self.cells_side_count_longitude,
+                                                  self.data_path.get(),
+                                                  self.point_side_size
+                                                  )
+
+            start_simulation(Neighbourhood.MOORE if self.neighborhood_var.get() == 0 else Neighbourhood.VON_NEUMANN,
+                             window)
+
+        def load_and_crop_image(self):
+            w, h = self.loaded_img.size
+
+            longitude_west_bound = int(w * (self.left_coord + 180) / 360)
+            longitude_east_bound = int(w * (self.right_coord + 180) / 360)
+            latitude_upper_bound = int(h * -(self.top_coord - 90) / 180)
+            latitude_lower_bound = int(h * -(self.down_coord - 90) / 180)
+
+            # TODO Temporary measure in place for the view to actually show anything when the coordinates values are very similiar
+            if longitude_west_bound + 10 >= longitude_east_bound:
```

Comment:
magic values - lets name it - Minimal preview size or something

Reply:
While I agree, this solution is only temporary assuming that preview will use map with greater level of detail

Reply:
ok